### PR TITLE
GLua eradication

### PR DESCRIPTION
**Pull request by oldmud0**
The purpose of this PR is to eradicate all occurrences of unofficial Lua syntax. All of it. This is an ongoing PR, and it will not merge until luacheck passes with no syntax errors across the entire source code.

### CODE REVIEW

**Comment by Divran** [comment]
I see you're even erasing brackets around if statements. That's not unofficial lua syntax, just sayin

File: lua/wire/wireshared.lua [hunk 1]
```diff
@@ -789,7 +789,7 @@ end
 
 	Very useful for searching algorithms
 	Used by custom spawn menu search & gate tool search, for example
-	Credits go to: http://lua-users.org/lists/lua-l/2009-07/msg00461.html
+	Credits go to: http:-- lua-users.org/lists/lua-l/2009-07/msg00461.html
```
Comment (AbigailBuccaneer) [inline lua/wire/wireshared.lua]: Please fix this and any other instances of incorrect replacements.

**Comment by oldmud0** [comment]
The parentheses thing I considered part of the "unofficial syntax" because Lua neither requires nor condones parentheses around the entire condition. It's annoying, it's not part of the manual of style, and it's not in new code.

I had to merge against master to fix a conflict with master branch that would not have allowed merging (like that's going to happen anytime soon). 

I was expecting someone to point out some mistakes I inevitably made trying to fix literally everything. Thank you for reviewing.

File: lua/entities/gmod_wire_button.lua [hunk 2]
```diff
@@ -99,12 +99,12 @@ end
 function ENT:Think()
 	self.BaseClass.Think(self)
 
-	if ( self:GetOn() ) then
+	if  self:GetOn()  then
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_button.lua]: there are extra spaces here

File: lua/entities/gmod_wire_cd_ray.lua [hunk 3]
```diff
@@ -97,41 +97,41 @@ function ENT:Setup(Range,DefaultZero)
...
 		self.WriteBuffer[0] = value
 	end
 end
 
 function ENT:DoJob()
-	if (not self.Disk) then return end
+	if not self.Disk then return end
 	local disk = self.Disk
-	if (self.Command[8] ~= 0) then
+	if self.Command[8] ~= 0 then
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_cd_ray.lua]: wow someone should really replace those numbers by constants with telling names

File: lua/wire/stools/trigger.lua [hunk 4]
```diff
@@ -87,7 +87,7 @@ function TOOL:RightClick( tr )
 		local ent = tr.Entity
 		if ent:GetClass() == "gmod_wire_trigger" then
 
-			-- http:--youtu.be/RTR1ny0O_io
+			-- http://youtu.be/RTR1ny0O_io
```
Comment (TomyLobo) [inline lua/wire/stools/trigger.lua]: another broken url

File: lua/wire/tool_loader.lua [hunk 5]
```diff
@@ -426,10 +426,10 @@ function WireToolSetup.setToolMenuIcon( icon )
...
 
 -- makes a new TOOL
---  s_mode: Tool_mode, same as the old tool lua file name, minus the "wire_" part
---  s_name: Proper name for the tool
---  s_class: For tools that make a device. Should begin with "gmod_wire_". Can be nil if not using WireToolObj.LeftClick or WireToolSetup.BaseLang
---  f_toolmakeent: Server side function for making the tools device. Can be nil if not using WireToolObj.LeftClick
+-- s_mode: Tool_mode, same as the old tool lua file name, minus the "wire_" part
+-- s_name: Proper name for the tool
+-- s_class: For tools that make a device. Should begin with "gmod_wire_". Can be nil if not using WireToolObj.LeftClick or WireToolSetup.BaseLang
+-- f_toolmakeent: Server side function for making the tools device. Can be nil if not using WireToolObj.LeftClick
```
Comment (TomyLobo) [inline lua/wire/tool_loader.lua]: Can you undo this change? I think the extra indentation here had a purpose.

**Comment by TomyLobo** [comment]
still working on the script.
turns out determining the difference between code and prose is hard.

File: lua/entities/gmod_wire_hudindicator/init.lua [hunk 6]
```diff
@@ -363,11 +363,11 @@ function ENT:Think()
 	return true
 end
 
-// Advanced Duplicator Support
+-- Advanced Duplicator Support
 function ENT:BuildDupeInfo()
 	local info = self.BaseClass.BuildDupeInfo(self) or {}
 
-	if (self.Pod) and (self.Pod:IsValid()) then
+	if self.Pod) and (self.Pod:IsValid() then
```
Comment (oldmud0) [inline lua/entities/gmod_wire_hudindicator/init.lua]: This was very hard to parse. I had to check each occurrence manually before replacing and make sure this didn't happen.

File: lua/entities/gmod_wire_egp/lib/egplib/textlayouter.lua [hunk 7]
```diff
@@ -1,48 +1,48 @@
-//----------------------------------------------------------------------------------------------
-//	Purpose: contains functionality for wrapping text to a specific width and expanding spaces
-//	so the text is aligned on both sides
-//----------------------------------------------------------------------------------------------
+------------------------------------------------------------------------------------------------
+-- 	Purpose: contains functionality for wrapping text to a specific width and expanding spaces
+-- 	so the text is aligned on both sides
```
Comment (oldmud0) [inline lua/entities/gmod_wire_egp/lib/egplib/textlayouter.lua]: sneaky tabs 

Sneaky, sneaky tabs...

File: lua/entities/gmod_wire_hdd.lua [hunk 8]
```diff
@@ -27,8 +27,8 @@ function ENT:Initialize()
 	self.Out = 0
 
 	-- Flash type
-	--   0: compatibility 16 values per block mode
-	--   1: 128 values per block mode
+	--  0: compatibility 16 values per block mode
+	--  1: 128 values per block mode
```
Comment (oldmud0) [inline lua/entities/gmod_wire_hdd.lua]: Not really, but ok

File: lua/wire/stools/trigger.lua (hunk 4 above)
Comment (oldmud0) [inline lua/wire/stools/trigger.lua]: Meant to fix this, but you beat me to it.

File: lua/wire/tool_loader.lua (hunk 5 above)
Comment (oldmud0) [inline lua/wire/tool_loader.lua]: If you see the previous occurrence of this, you will see that the in-comment indentation is a bit inconsistent. This one only has space of indentation whereas the previous one had two.

**Comment by oldmud0** [comment]
Thanks for your contributions. I wanted to do some things too but it looks like you have beat me to them. I'll do the second pass now.

File: lua/entities/gmod_wire_expression2/core/string.lua [hunk 9]
```diff
@@ -1,28 +1,25 @@
...
-// TODO: these return bad results when used with negative numbers!
-// TODO: benchmarks!
+-- TODO: is string.left() faster than s:left()?
+-- TODO: is string.sub faster than both left and right?
+-- TODO: these return bad results when used with negative numbers!
+-- TODO: benchmarks!
 
 local string = string -- optimization
 
-/******************************************************************************/
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_expression2/core/string.lua]: maybe reinstate those as lines of dashes?

File: lua/entities/gmod_wire_expression2/core/cl_files.lua [hunk 10]
```diff
@@ -48,10 +48,10 @@ local function process_filepath( filepath )
 	return string.GetPathFromFilename( fullpath ) or "e2files/", string.GetFileFromFilename( fullpath ) or "noname.txt"
 end
 
-/* --- File Read --- */
+--[[ --- File Read --- ]]
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_expression2/core/cl_files.lua]: could have just removed the /* */ :D

File: lua/entities/gmod_wire_expression2/core/string.lua (hunk 9 above)
Comment (oldmud0) [inline lua/entities/gmod_wire_expression2/core/string.lua]: I thought I did

File: lua/entities/gmod_wire_expression2/core/cl_files.lua (hunk 10 above)
Comment (oldmud0) [inline lua/entities/gmod_wire_expression2/core/cl_files.lua]: You're right, that would have looked a lot neater.

**Comment by oldmud0** [comment]
Okay, so I have found another problem. When `e2function`s are defined, the declaration is considered an error by luacheck, so basically any file defining any E2 functions is an error (afaik luacheck stops at the first error on a file).

File: lua/entities/gmod_wire_relay.lua [hunk 11]
```diff
@@ -44,23 +44,23 @@ function ENT:Setup(keygroup1, keygroup2, keygroup3, keygroup4, keygroup5, keygro
...
 	self.nokey			= nokey
 	
-	//build inputs and putputs, init all nil values
+	-- build inputs and putputs, init all nil values
 	for p=1, self.poles do
 		self.outputs[p] = outpoles[p]
 		self.Value[p] = self.Value[p] or 0
 		for t=1, self.throws do
-			//inputs[ p * self.poles + t ] = t .. outpoles[p]
+			-- inputs[ p * self.poles + t ] = t .. outpoles[p]
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_relay.lua]: i use "--" without a space for commented out code and "--" with a space for the rest

**Comment by TomyLobo** [comment]
re e2function: Can you maybe preprocess them with a regex before piping them into luacheck?
I could give you a sed script that turns e2functions into regular functions, probably even with all the parameter names intact.

re continue: maybe you can do the same here... replace it by break.

**Comment by oldmud0** [comment]
As a conclusion, the effort was noble, but is not easily doable because it involves touching some really old code that is known to work and also reflowing code that depends on `continue`.

If I wanted to do the eradication, I'd probably just start over, scout for common style problems, do the heavy lifting purely with regex/awk, and then do the final pass by hand. I made too many mistakes trying to do this with semi-manual regex, most especially just missing things by going too fast.
